ls1intum/Artemis
# [Exam] Allow student complaints in the student exam summary

### Checklist

### Motivation and Context
We want to give the students the opportunity to request more feedback or to complain about an assessment. 

### Description
Insert the complaint interaction component in the exam summary page.
Propagate the required inputs and handle the StudentReviewPeriod

**It is intended behaviour, that you cannot see the complaints of your assessment.** Only other tutors can react to complaints about your assessment. To view the complaint you need to log on with another account which has tutor rights

### Steps for Testing
1. Log in to Artemis
2. Create an exam and add **all the dates for the exam.** 
3. Add different exercises, for programing exercises add a due date & check the manual assessment option)
4. Take part in the exam 
5. Assess the exam exercises with a different account
6. log back in as a student and view the summary page (after student review begin)
7. complain/request feedback. 
8. Interact with the requests as a tutor (in the exam tutor dashboard)

### Screenshots
**Exam with complain enabled**
![image](https://user-images.githubusercontent.com/15110960/87556421-ebc31680-c6b6-11ea-8c8e-0d544cc84441.png)

----------------------------------------------------------------------------------

**Text Exercise: complaint rejected**
![image](https://user-images.githubusercontent.com/15110960/87556843-7146c680-c6b7-11ea-9fad-acc1e4376183.png)

**Programming exercise Feedback given
![image](https://user-images.githubusercontent.com/15110960/87556451-f67dab80-c6b6-11ea-9cb9-5a5627e9d5e9.png)

**Modeling Exercise complaint accepted**
![image](https://user-images.githubusercontent.com/15110960/87556499-05645e00-c6b7-11ea-82b8-d9cbca8c3d34.png)
